Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import class Foundation.NSLock
+import protocol Foundation.LocalizedError
+import struct Foundation.Data  // only for convenience initializers
+
+/// The type representing a request or response body.
+public final class Body: @unchecked Sendable {
+
+    /// The underlying data type.
+    public typealias DataType = ArraySlice<UInt8>
+
+    /// How many times the provided sequence can be iterated.
+    public enum IterationBehavior: Sendable {
+
+        /// The input sequence can only be iterated once.
+        ///
+        /// If a retry or a redirect is encountered, fail the call with
+        /// a descriptive error.
+        case single
+
+        /// The input sequence can be iterated multiple times.
+        ///
+        /// Supports retries and redirects, as a new iterator is created each
+        /// time.
+        case multiple
+    }
+
+    /// How many times the provided sequence can be iterated.
+    public let iterationBehavior: IterationBehavior
+
+    /// The total length of the body, if known.
+    public enum Length: Sendable {
```

Comment:
Same here. Probably a `struct`

Reply:
Because API evolution? If so, I'd happily mark this as `@frozen` or give some other hint that we won't be expanding the enum.